internal/gen/messages_semantical.go

```diff
@@ -0,0 +1,90 @@
+// Copyright © 2021 The Things Industries B.V.
+// SPDX-License-Identifier: Apache-2.0
+
+package gen
+
+import (
+	"github.com/TheThingsIndustries/protoc-gen-go-flags/annotations"
+	"google.golang.org/protobuf/compiler/protogen"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+func (g *generator) messageHasSemanticalFlags(message *protogen.Message) bool {
+	// No code is generated for map entries, so we also don't need to generate semantical flags.
+	if message.Desc.IsMapEntry() {
+		return false
+	}
+
+	var generateSemanticalFlags bool
+
+	for _, field := range message.Fields {
+		// If the field has the (thethings.flags.field) option, and semantical is set, we need to generate semantical flags for the message.
+		fieldOpts := field.Desc.Options().(*descriptorpb.FieldOptions)
+		if proto.HasExtension(fieldOpts, annotations.E_Field) {
```

Comment:
You can save some indentation here by doing `if !condition { continue }` instead of nesting.

Reply:
I agree that that is cleaner, just kept them consistent with the `set` and `select` versions.